src/openssl/lib_crypto.cr

```diff
@@ -1,5 +1,15 @@
-@[Link("crypto")]
+ifdef darwin
+  @[Link("crypto")]
+else
+  @[Link(ldflags: "`pkg-config --libs libcrypto`")]
+end
 lib LibCrypto
+  ifdef darwin
+    OPENSSL_102 = false
+  else
+    OPENSSL_102 = {{`(pkg-config --atleast-version=1.0.2 libcrypto && echo true) || echo false`.id}}
```

Comment:
Note: darwin may have OpenSSL 1.0.2 installed (brew install `openssl`) and it's perfectly possible to link against it `--link-flags="-L/usr/local/opt/openssl/lib"`.